thunder/core/transforms.py

```diff
@@ -3121,7 +3123,9 @@ def deconstruct_forward_env_for_backward(trace, env):
     # arguments. See test_grad.py:test_torch_autograd_function for an example
     # where this is tested.
     bound_symbols = iter_bound_symbols(trace.bound_symbols)
-    saved_for_backward = tuple(env[sequencify(symbol.output)[0].name].residuals for symbol in bound_symbols)
+    saved_for_backward = tuple(
+        env[sequencify(symbol.output)[0].name].residuals for symbol in bound_symbols if symbol.output is not None
+    )
```

Comment:
Could you please revert this change and the one in `reconstruct_forward_env_for_backward`?
Instead, we can modify the common function `iter_bound_symbols` to skip symbols with None output.